### DISCUSSION TITLE
Flet takes too long to boot up the application

### DISCUSSION THREAD

**Discussion by xKristee29** [flet-dev/flet#3203] [Q&A] [answered]
I have a problem related to the long startup of an application in flet. It could take even 5 minutes to show a screen, and after that it crashes, but when I destroy the process and rerun it it takes only 2-3 seconds to start the app.

Can anyone explain how can the first startup of an app take so long?

**Comment by RRozi** [comment]
This usually always happens with all sites, just not for so long, but everything takes a little longer the first time, this happens because the entire site is loaded on the client for the first time - the first time it saves everything it needs in cookies (by the way, it doesn’t ask about this) But and the next time you log in, it simply pulls up everything from the cookies.

THIS IS MY PERSONAL SUBJECTIVE OPINION HOW THIS HAPPENS

**Comment by xKristee29** [comment, marked as answer]
I found the problem and I think it could be helpful for big projects.

When I was running the project I used the ```flet -r``` command, that is indexing all the project's file tree. So by growing the file quantity we extend the time it takes to startup the app..

As a solution I recommend avoiding running the ```flet -r``` command and opt out for:
```
flet main.py
---------- OR --------
python main.py
```

**Reply by Krishna-Noutiyal** [reply]
I have the same issue with my windows software, the app renders any view super slow. But I don't understand what does opting out from `flet -r` mean. I compile the software with `flet build` and the compiled software takes around 1 min to render in any view. How can not using `flet -r` or `flet run` have any impact on the compiled application.

Also I found one interesting thing, when I compile a standalone using `flet pack` the compile exe runs smoothly without any boot delay. But when I compile using `flet build` the boot time of the compiled software is increased a lot.

**Comment by mquasar** [comment]
I'm running into the same issue. 

No matter how I run the app (a very simple one, with only two text fields and a button), it takes more than 7 minutos to run, even after the first run. 

It had been running normally, but for some reason now it takes forever to start the little app. 

I'm using uv and python 3.13.3 on Windows.

I have deleted the project folder and recreated it, but the slowness endures.
